tests/v2/test_v2_rpc.py

```diff
@@ -0,0 +1,97 @@
+import importlib
+import logging
+from collections import deque
+from datetime import timedelta
+from typing import (
+    Literal,
+)
+
+import pytest
+from pytest_snapshot.plugin import Snapshot
+
+from replit_river.v2.client import Client
+from tests.fixtures.codegen_snapshot_fixtures import validate_codegen
+from tests.v2.datagrams import (
+    ClientId,
+    FromClient,
+    ServerId,
+    StreamAlias,
+    TestTransport,
+    ToClient,
+    ValueSet,
+    WaitForClosed,
+)
+
+logger = logging.getLogger(__name__)
+
+_AlreadyGenerated = False
+
+
+@pytest.fixture(scope="function", autouse=True)
+def stream_client_codegen(snapshot: Snapshot) -> Literal[True]:
+    global _AlreadyGenerated
+    if not _AlreadyGenerated:
+        validate_codegen(
+            snapshot=snapshot,
+            snapshot_dir="tests/v2/codegen/snapshot/snapshots",
+            read_schema=lambda: open("tests/v2/test_rpc.schema.json"),
+            target_path="test_basic_rpc",
+            client_name="StreamClient",
+            protocol_version="v2.0",
+        )
+        _AlreadyGenerated = True
```

Comment:
This flag is _probably_ not necessary. Tests are executed sequentially, so even though we'd be overwriting the client with itself, there would never be in-process writes.